### ISSUE TITLE
Something about specifying the GPU

### ISSUE THREAD

**Issue by yongrenr** [kuleshov-group/caduceus#7]
Hello, I have a question for you. Due to the limited number of devices on our side, I want to specify the GPU, but I still can't change the environment after setting, so where do I specify the GPU?

**Comment by yair-schiff** [comment]
Hi @yongrenr, are you asking about specifying the GPU type or the number of GPUs/devices for the training runs. To change the number of devices, you add the following CLI flag: `trainer.devices=<NUM_GPUS>`. See [here](https://github.com/kuleshov-group/caduceus/blob/6ebb434f55b2ee5dab6e6f55c41581b113786dd9/slurm_scripts/run_pretrain_caduceus.sh#L56C3-L56C18), for example.

Please let me know if this resolves the confusion.

**Comment by yongrenr** [comment]
Thanks for the reminder, it's working normally for now

**Comment by yongrenr** [comment]
I also want to ask if I divide the data set into 2 parts, a test set and a training set. By training your model and then testing it, can I feed 2 data sets at the same time, or can I train with the training set first and then use ckpt to test on the test data?

**Comment by yair-schiff** [comment]
You can definitely pass a train and test set at the same time. Take a look at this [file](https://github.com/kuleshov-group/caduceus/blob/main/src/dataloaders/genomics.py) to see how this is done for other datasets we use. You can hopefully pattern match from these to your dataset. Let me know if you’re having any issues.

**Comment by yongrenr** [comment]
Thank you for your reminder. I will try it. If I don’t understand something, please give me some advice.
